components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOServiceProviderDAO.java

```diff
@@ -200,34 +206,23 @@ public boolean addServiceProvider(SAMLSSOServiceProviderDO serviceProviderDO)
             }
 
             Resource resource = createResource(serviceProviderDO);
-            try {
-                if (!isTransactionStarted) {
-                    registry.beginTransaction();
-                }
-
-                registry.put(path, resource);
-
-                if (!isTransactionStarted) {
-                    registry.commitTransaction();
-                }
-
-            } catch (RegistryException e) {
-                if (!isTransactionStarted) {
-                    registry.rollbackTransaction();
-                }
-                throw e;
+            if (!isTransactionStarted) {
+                registry.beginTransaction();
             }
-
+            registry.put(path, resource);
+            if (log.isDebugEnabled()) {
+                log.debug("Service Provider " + serviceProviderDO.getIssuer()
+                        + " is added successfully.");
+            }
+            return true;
         } catch (RegistryException e) {
-            log.error("Error While adding Service Provider", e);
-            throw IdentityException.error("Error while adding Service Provider", e);
-        }
-
-        if (log.isDebugEnabled()) {
-            log.debug("Service Provider " + serviceProviderDO.getIssuer()
-                    + " is added successfully.");
+            isErrorOccurred = true;
+            String msg = "Error while adding Service Provider";
```

Comment:
Can we pit the Issuer on the error, since it is known and not null at this point.